ios/engine/KMEI/KeymanEngine/Classes/LanguagePicker/LanguageViewController.swift

```diff
@@ -196,20 +201,20 @@ class LanguageViewController: UITableViewController, UIAlertViewDelegate {
     }
   }
 
-  @objc func languageFetchFinished() {
+  private func languagesUpdated() {
     dismissActivityView()
     tableView.reloadData()
     if numberOfSections(in: tableView) == 0 {
       showConnectionErrorAlert()
     }
   }
 
-  @objc func languageFetchFailed() {
+  private func languagesDownloadFailed(_ notification: LanguagesDownloadFailedNotification) {
```

Comment:
Could we maybe change the parameter name to something more appropriate?  We aren't receiving Notification objects, are we?  I get that there's some type aliasing going on, but it's quite confusing to see a post for this notification taking an error, but the handling methods are receiving some sort of "Notification" object, even if that Notification name is just a type alias.

There are other places in this PR that do similar things for other notification names.  Take this as a general request, rather than one specific to the `languagesDownloadFailed` stuff.

Reply:
In the places where the notification value is actually used, the parameter name is usually more descriptive. I think `notification` is an adequate name where the value is unused since the function is happy to take some abstract notification value. The parameter is also less prone to having the wrong name if the notification type is changed in the future. A generic name of `notification` would still be correct but perhaps the value would no longer being correctly described by a name such as `error`.

Also in Xcode, it's easy to inspect the underlying type beneath the typealias so it's pretty clear it's not `Foundation.Notification`.

Reply:
Added functionality to permit the function to be void if it doesn't need the value so most of these should disappear.